app/common/directives/modal.js

```diff
@@ -26,9 +28,10 @@ function (
             $scope.iconPath = $scope.icon ? '../../img/iconic-sprite.svg#' + $scope.icon : '';
 
             var classChangePromise = null;
-
+            $scope.iconPath = '../../img/iconic-sprite.svg#' + $scope.icon;
             $scope.classVisible = false;
             $scope.modalOffset = 0;
+            $scope.title = $translate.instant($scope.title);
```

Comment:
Hrm..  If it worked with the promise thats fine, but the docs for ngtranslate didn't seem like it work. ngResource does some special magic with its promises that makes this sort thing work on endpoints, but I thought that was a special case.